src/ansible_compat/runtime.py

```diff
@@ -0,0 +1,89 @@
+"""Ansible runtime environment maanger."""
+import os
+import subprocess
+from typing import TYPE_CHECKING, Any, List, Optional, Union
+
+from packaging.version import Version
+
+from ansible_compat.config import AnsibleConfig, parse_ansible_version
+from ansible_compat.errors import MissingAnsibleError
+from ansible_compat.prerun import get_cache_dir
+
+if TYPE_CHECKING:
+    # https://github.com/PyCQA/pylint/issues/3240
+    # pylint: disable=unsubscriptable-object
+    CompletedProcess = subprocess.CompletedProcess[Any]
+else:
+    CompletedProcess = subprocess.CompletedProcess
+
+
+class Runtime:
+    """Ansible Runtime manager."""
+
+    _version: Optional[Version] = None
+    cache_dir: Optional[str] = None
+
+    def __init__(
+        self, project_dir: Optional[str] = None, isolated: bool = False
+    ) -> None:
+        """Initialize Ansible runtime environment.
+
+        Isolated mode assures that installation of collections or roles
+        does not affect Ansible installation, an unique cache directory
+        being used instead.
+        """
+        self.project_dir = project_dir or os.getcwd()
+        if isolated:
+            self.cache_dir = get_cache_dir(self.project_dir)
+        self.config = AnsibleConfig()
+
+    # pylint: disable=no-self-use
+    def exec(self, args: Union[str, List[str]]) -> CompletedProcess:
```

Comment:
I'd rather have a `check_call()` style thingy. Plus we need to address the naming concern I  raised above.